### PR TITLE
Course passed if has course run certificate

### PR DESCRIPTION
**Pull request by annagav**
# What are the relevant tickets?
Fix https://github.com/mitodl/hq/issues/2319

# Description (What does it do?)
Don't show green label "Course Passed" if there is no certificate.

# Screenshots (if appropriate):
This green label "Course Passed" shouldn't show, if there is no certificate.
Screen Shot 2023-09-11 at 3 33 30 PM

# How can this be tested?
create a passing grade for user and a revoked coure run certificate. Check the dashboard course card enrollment info. It should not show the certificate and not show the "Course Passed" tag.

### CODE REVIEW

File: frontend/public/src/components/EnrolledItemCard_test.js [hunk 1]
```diff
@@ -147,41 +147,39 @@ describe("EnrolledItemCard", () => {
...
-    grade.passed = false
+    grade.passed = true
     enrollmentCardProps.enrollment.grades = [grade]
     inner = await renderedCard()
     coursePassed = inner.find(".enrolled-item").find(".badge-enrolled-passed")
     assert.isFalse(coursePassed.exists())
   })
   ;[true, false].forEach(isSelfPaced => {
-    it("renders the course passed label based on self_paced and course passed", async () => {
+    it("renders the course passed label based on course certificate independent of course pacing", async () => {
```
Comment (collinpreston) [inline frontend/public/src/components/EnrolledItemCard_test.js:167]: What is this specifically testing for?

File: frontend/public/src/components/EnrolledItemCard_test.js [hunk 2]
```diff
@@ -147,41 +147,39 @@ describe("EnrolledItemCard", () => {
...
-        assert.isTrue(coursePassed.exists())
-      } else {
-        assert.isFalse(coursePassed.exists())
-      }
+      assert.isTrue(coursePassed.exists())
     })
   })
   ;[true, false].forEach(past => {
-    it("renders the course passed label based on course end and certificate available dates in past", async () => {
+    it("doesn't render the course passed label based on course end and certificate available dates in past", async () => {
```
Comment (collinpreston) [inline frontend/public/src/components/EnrolledItemCard_test.js:182]: Is this test needed in order to enforce a requirement?

**Review by collinpreston — CHANGES_REQUESTED** [review]
Testing worked for me.  Couple comments and a suggestion.

Should we rename https://github.com/mitodl/mitxonline/blob/c2dc6f58fbd12248e47db4715c8ce88ad0ce90bd/frontend/public/src/lib/courseApi.js#L120 ?

File: frontend/public/src/components/EnrolledItemCard_test.js (hunk 1 above)
Comment (annagav) [reply frontend/public/src/components/EnrolledItemCard_test.js:167]: It checks that the "course passed" label is shown independently of whether the course is self paced or not. But I don't know the history of why that matters. Do you have any ideas?

Comment (collinpreston) [reply frontend/public/src/components/EnrolledItemCard_test.js:167]: I don't think this should exist if the label is independent of whether the course is self-paced or not.

File: frontend/public/src/components/EnrolledItemCard_test.js (hunk 2 above)
Comment (collinpreston) [reply frontend/public/src/components/EnrolledItemCard_test.js:182]: I agree that this should be removed.

**Review by collinpreston — APPROVED** [review]
Nice, looks good to me!

**Pull request merged**